include/ceed.h

```diff
@@ -530,6 +530,8 @@ CEED_EXTERN int CeedOperatorLinearAssemblePointBlockDiagonal(CeedOperator op,
     CeedVector assembled, CeedRequest *request);
 CEED_EXTERN int CeedOperatorLinearAssembleAddPointBlockDiagonal(CeedOperator op,
     CeedVector assembled, CeedRequest *request);
+CEED_EXTERN int CeedOperatorLinearFullAssemble(CeedOperator op,
+    CeedInt *nentries, CeedInt **rows, CeedInt **cols, CeedScalar **vals);
```

Comment:
Would you be willing to split this into `CeedOperatorLinearAssembleSymbolic(CeedOperator, size_t *nentries, CeedInt **rows, CeedInt **cols)` and `CeedOperatorLinearAssemble(CeedOperator, CeedVector values)`?

My rationale is that we often assemble multiple times using the same nonzero pattern. Many backends can use the same Symbolic code. The indices can live on the preferred `CeedMemType` or we can have another argument to specify where they go. We support this split phase COO assembly on GPUs in PETSc and one can always call both functions if they need both.

If we're going to store the entries in a `CeedVector` (which seems sensible because it conveniently works with both host and device) then its constructor `CeedVectorCreate` needs to use `size_t` instead of `CeedInt`.

Reply:
I was going to chat with @barker29 on Monday about possible directions for this code. It sounds like maybe all three of us should chat?

Reply:
Sure, though I suspect the interface can be fleshed out here. @stefanozampini added the split-phase assembly to PETSc, and is familiar with MFEM.

Reply:
My concern is less on the interface is and more in the internals. As is, this PR introduces code duplication from our (albeit complicated) code for diagonal assembly. We pushed that code into the backed 'families' for performance and differences in qdata layout. I'm trying to understand this intended use case to see if the performance concerns apply here. I think the diagonal assembly code can be extended to do this without the duplication, but I'm still fuzzy on how this data will be consumed.

Reply:
I can certainly split the function into two as @jedbrown suggests, though that will require some significant changes to the current implementation. Using a `CeedVector` for the values is easy.

@jeremylt 's concerns about code duplication with the diagonal assembly are a bit harder for me to understand what's actually wanted but we can hash that out in our meeting today.